planner/core/logical_plans.go

```diff
@@ -142,6 +142,7 @@ type LogicalJoin struct {
 	preferJoinOrder bool
 
 	EqualConditions []*expression.ScalarFunction
+	NAEQConditions  []*expression.ScalarFunction
```

Comment:
Will it be better if we use a bool value to tell whether is null aware or not?

Reply:
// Step2: when step1 is finished, then we can determine whether we need to extract NA-EQ from OtherCondition to NAEQConditions.
// when there are still no EqualConditions, let's try to be a NAAJ.

In where the comment is, we only fill the NAEQConditions when there are no common EqualConditions.